python/pyiceberg/io/pyarrow.py

```diff
@@ -725,6 +729,22 @@ def primitive(self, primitive: pa.DataType) -> IcebergType:
         raise TypeError(f"Unsupported type: {primitive}")
 
 
+def _starmap_task_to_table(
```

Comment:
Same here, maybe better to let `_task_to_table` accept a single tuple argument, and then we can directly expand those.